### PR TITLE
ref(subscriptions): Decouple subscription logic from datasets [INGEST-627]

### PR DESCRIPTION
**Pull request by ahmedetefy**
Decouples subscription creation and deletion logic from datasets to also consider entities since dataset to entity is not a 1-to-1 relation

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -2169,7 +2169,14 @@ def test_error_handler(self, pipeline_mock: MagicMock) -> None:
...
 
-    def test(self) -> None:
+    @pytest.mark.parametrize(
+        "url",
+        [
+            "events/subscriptions",  # Only dataset in url
+            "events/events/subscriptions",  # dataset and entity in url
+        ],
+    )
+    def test(self, url: str) -> None:
```
Comment (evanh) [inline tests/test_api.py:2179]: Where is `url` used in this test?

**Review by evanh — COMMENTED** [review]
Can you add a test that ensures that the entity being specified is the one being used, and not just the default for the dataset? So for example use dataset `discover` but entity `transactions` or something like that.

File: tests/test_api.py (hunk 1 above)
Comment (ahmedetefy) [reply tests/test_api.py:2179]: oops .. I missed that

**Comment by ahmedetefy** [comment]
@evanh I have addressed the PR comments.. Let me know if there is anything else :)

**Review by evanh — APPROVED** [review]
LGTM. In order to use the `--entity` option in prod you'll need to make an ops PR as well.

**Comment by ahmedetefy** [comment]
> LGTM. In order to use the `--entity` option in prod you'll need to make an ops PR as well.

Thanks! Can you tell me where to edit that?

**Pull request merged**